t/Mockify_StaticInjection.t

```diff
@@ -84,63 +95,145 @@ sub test_MethodAndImportedFunctionHaveTheSameName {
-    $Mockify->spy('overrideMethod_spy')->when(String('spyme'));
-    my $SUT = $Mockify->getMockObject();
+
+    my $injector = Test::Mockify::Injector->new('FakeStaticTools');
+    $injector->mock('ReturnHelloWorld')->when(String('Error'))->thenThrowError('TestError');
+    $injector->mock('ReturnHelloWorld')->when(String('caller'))->thenCall(sub{return 'returnValue'});
+    $injector->mock('ReturnHelloWorld')->when(String('undef'), String('abc'))->thenReturnUndef();
+
+    my $injector2 = Test::Mockify::Injector->new('FakeModuleStaticInjection');
+    $injector2->mock('overrideMethod')->when(String('override'))->thenReturn('overridden Value');
+    $injector2->spy('overrideMethod_spy')->when(String('spyme'));
 
     #Error
-    throws_ok( sub{$SUT->useStaticFunction('Error') },
+    throws_ok( sub{FakeModuleStaticInjection->useStaticFunction('Error') },
                    qr/TestError/,
                    "$SubTestName - prove if thenThrowError will fail"
     );
     # anonymous function pointer mock
-    is($SUT->useStaticFunction('caller'), 'caller: returnValue',"$SubTestName - prove thenCall");
+    is(FakeModuleStaticInjection->useStaticFunction('caller'), 'caller: returnValue', "$SubTestName - prove thenCall");
     # undef
-    is($SUT->useStaticFunction('undef', 'abc'), 'undef: ',"$SubTestName - prove thenReturnUndef");
+    is(FakeModuleStaticInjection->useStaticFunction('undef', 'abc'), 'undef: ', "$SubTestName - prove thenReturnUndef");
+
+    # TODO: Fix. For some reason it starts failing here.
     # normal mock
-    is($SUT->overrideMethod('override'), 'overridden Value',"$SubTestName - prove mock Works");
+    is(FakeModuleStaticInjection->overrideMethod('override'), 'overridden Value', "$SubTestName - prove mock Works");
     # normal spy
-    $SUT->overrideMethod_spy('spyme'); #1
-    $SUT->overrideMethod_spy('spyme'); #2
-    is(GetCallCount($SUT,'overrideMethod_spy'), 2,"$SubTestName - prove verify works for override");
+    FakeModuleStaticInjection->overrideMethod_spy('spyme'); #1
+    FakeModuleStaticInjection->overrideMethod_spy('spyme'); #2
+    is(GetCallCount($injector2->getVerifier(), 'overrideMethod_spy'), 2, "$SubTestName - prove verify works for override");
+}
+#----------------------------------------------------------------------------------------
+sub test_mockRevertsWhenInjectorGoesOutOfScope {
+    my $self = shift;
+    my $SubTestName = (caller(0))[3];
+
+    my $originalValue = FakeModuleForMockifyTest::DummyMethodForTestOverriding();
+    my $originalValue2 = FakeModuleForMockifyTest::secondDummyMethodForTestOverriding();
+    my $mockValue = 'A mock dummy method';
+    my $mockValue2 = 'A second mock dummy method';
+    my $verifier;
+
+    {
+        my $injector = Test::Mockify::Injector->new('FakeModuleForMockifyTest');
+        $injector->mock('DummyMethodForTestOverriding')->when()->thenReturn($mockValue);
+        $injector->addMock('secondDummyMethodForTestOverriding', sub { $mockValue2 });
```

Comment:
addMock is deprecated. Use  $injector->mock('secondDummyMethodForTestOverriding')->whenAny()->thenCall(sub {});
http://search.cpan.org/~chrisbr/Test-Mockify-1.3/lib/Test/Mockify/ReturnValue.pm#thenCall

Reply:
I understand that addMock is deprecated. I intentionally included it in this test, because the Injector was not initially backwards-compatible with the deprecated methods. With my last round of changes, now it is. I suppose the Injector doesn't really need to be, and I see you created an Issue to remove them at some point. I was just trying to keep the Injector as close to Mockify as possible.